pkg/pipeline/transform/transform_network.go

```diff
@@ -39,17 +39,17 @@ type Network struct {
 	api.TransformNetwork
 }
 
-func (n *Network) Transform(inputEntries []config.GenericMap) []config.GenericMap {
+func (n *Network) Transform(input []config.GenericMap) []config.GenericMap {
 	outputEntries := make([]config.GenericMap, 0)
-	for _, entry := range inputEntries {
+	for _, entry := range input {
 		outputEntry := n.TransformEntry(entry)
 		outputEntries = append(outputEntries, outputEntry)
 	}
 	return outputEntries
 }
 
 func (n *Network) TransformEntry(inputEntry config.GenericMap) config.GenericMap {
-	outputEntries := inputEntry
+	outputEntry := inputEntry.Copy()
```

Comment:
I would keep a comment every time we use `Copy()` to avoid future removal

```suggestion
	// copy input entry before transform to avoid alteration on parallel stages
	outputEntry := inputEntry.Copy()
```